### PR TITLE
Fixed geocoding on adherent updating profile

### PR DESCRIPTION
**Pull request by HeahDude**

### CODE REVIEW

File: src/AppBundle/Membership/MembershipRequestHandler.php [hunk 1]
```diff
@@ -57,7 +57,7 @@ public function update(Adherent $adherent, MembershipRequest $membershipRequest)
     {
         $adherent->updateMembership($membershipRequest, $this->addressFactory->createFromAddress($membershipRequest->getAddress()));
 
-        $this->manager->flush();
```
Comment (tgalopin) [inline src/AppBundle/Membership/MembershipRequestHandler.php]: Is it normal to remove the flush then?

Comment (HeahDude) [inline src/AppBundle/Membership/MembershipRequestHandler.php]: Yes it's done here https://github.com/EnMarche/en-marche.fr/blob/master/src/AppBundle/Geocoder/Subscriber/EntityAddressGeocodingSubscriber.php#L40.

Comment (hhamon) [inline src/AppBundle/Membership/MembershipRequestHandler.php]: What if the address didn't change? We're still regeocoding it and flushing no changes. Fortunately the geocoder in production uses a cache for geocoded addresses.

**Comment by hhamon** [comment]
Can you still add a unit test in the `EntityAddressGeocodingSubscriberTest` suite please?

**Comment by HeahDude** [comment]
@hhamon I've pushed 5eabc00, can you please check?

File: src/AppBundle/Entity/PostAddress.php [hunk 2]
```diff
@@ -179,4 +179,13 @@ public function getGeocodableAddress(): string
     {
         return (string) GeocodableAddress::createFromAddress($this);
     }
+
+    public function equals(self $other)
```
Comment (hhamon) [inline src/AppBundle/Entity/PostAddress.php]: Missing return type hint.

File: src/AppBundle/Entity/PostAddress.php [hunk 3]
```diff
@@ -179,4 +179,13 @@ public function getGeocodableAddress(): string
...
+    {
+        return $this->address !== $other->getAddress()
```
Comment (hhamon) [inline src/AppBundle/Entity/PostAddress.php]: I would better do something like

```php
public function equals(self $other): bool
{
    return mb_strtolower($this->address) === mb_strtolower($other->getAddress())
        && mb_strtolower($this->postalCode) === mb_strtolower($other->getPostalCode())
        && ....
    ;
}
```

Comment (hhamon) [inline src/AppBundle/Entity/PostAddress.php]: Don't forget to add some unit tests for the method as well ;)

Comment (HeahDude) [inline src/AppBundle/Entity/PostAddress.php]: Right, done in 7c35b5f, thanks!

File: src/AppBundle/Entity/PostAddress.php [hunk 4]
```diff
@@ -180,12 +180,12 @@ public function getGeocodableAddress(): string
...
     }
 
-    public function equals(self $other)
+    public function equals(self $other): bool
     {
-        return $this->address !== $other->getAddress()
-               || $this->cityName !== $other->getCityName()
-               || $this->postalCode !== $other->getPostalCode()
-               || $this->country !== $other->getCountry()
+        return mb_strtolower($this->address) !== mb_strtolower($other->getAddress())
```
Comment (hhamon) [inline src/AppBundle/Entity/PostAddress.php]: Why `||` ? It must be equal if all attributes are equal at the same time so you must use `&&`.

Comment (HeahDude) [inline src/AppBundle/Entity/PostAddress.php]: Yeah, wrong copy-paste... I've fixed when adding the missing tests ;) See 7aaff81.

File: tests/AppBundle/Entity/PostAddressTest.php [hunk 5]
```diff
@@ -20,4 +20,20 @@ public function testCreateFullFrenchAddress()
         $this->assertSame(5.987654, $address->getLongitude());
         $this->assertSame('92 bld Victor Hugo, 92110 Clichy, France', $address->getGeocodableAddress());
     }
+
+    public function testEquals()
+    {
+        $address = PostAddress::createFrenchAddress('92 bld Victor Hugo', '92110-92024');
```
Comment (hhamon) [inline tests/AppBundle/Entity/PostAddressTest.php]: `$address1`

File: tests/AppBundle/Entity/PostAddressTest.php [hunk 6]
```diff
@@ -20,4 +20,20 @@ public function testCreateFullFrenchAddress()
...
+        $address2 = PostAddress::createFrenchAddress('92 bld Victor Hugo', '92110-92024');
+
+        $this->assertTrue($address->equals($address2));
```
Comment (hhamon) [inline tests/AppBundle/Entity/PostAddressTest.php]:
```php
$this->assertTrue($address1->equals($address2));
$this->assertTrue($address2->equals($address1));
```

File: tests/AppBundle/Entity/PostAddressTest.php [hunk 7]
```diff
@@ -20,4 +20,20 @@ public function testCreateFullFrenchAddress()
...
+    }
+
+    public function testNotEquals()
+    {
+        $address = PostAddress::createFrenchAddress('92 bld Victor Hugo', '92110-92024');
```
Comment (hhamon) [inline tests/AppBundle/Entity/PostAddressTest.php]: `$address1`

File: tests/AppBundle/Entity/PostAddressTest.php [hunk 8]
```diff
@@ -20,4 +20,20 @@ public function testCreateFullFrenchAddress()
...
+        $address2 = PostAddress::createFrenchAddress('94 bld Victor Hugo', '92110-92024');
+
+        $this->assertFalse($address->equals($address2));
```
Comment (hhamon) [inline tests/AppBundle/Entity/PostAddressTest.php]:
```php
$this->assertFalse($address1->equals($address2));
$this->assertFalse($address2->equals($address1));
```

**Pull request merged**